torch/_inductor/codegen/cpp.py

```diff
@@ -741,7 +750,16 @@ def load(self, name: str, index: sympy.Expr):
         if expanded_index == new_index:
             line = f"at::vec::Vectorized<float>({var}[{cexpr(index)}])"
         else:
-            line = f"at::vec::Vectorized<float>::loadu({var} + {cexpr(new_index)})"
+            if V.graph.get_dtype(name) in [torch.bool, torch.uint8]:
+                g_tmp_buf = f"g_tmp_buffer_{var}"
+                nelements = codecache.pick_vec_isa().nelements()
+                self.loads.writeline(f"float {g_tmp_buf}[{nelements}] = {{0}};")
+                self.loads.writeline(
+                    f"flag_to_float({var} + {cexpr(new_index)}, {g_tmp_buf}, {nelements});"
+                )
```

Comment:
Is it possible that we have two loads on the same `var`? If so, there would be name clashing? Maybe safer to put these two lines into a local scope?

Reply:
Since we have passed TorchBench, HF, and TIMM, we have not encountered this case. Even if a tensor is a view of another tensor, there should be two different tensors with different variable names.